### PR TITLE
qutebrowser: add wrapQtAppsHook

### LINKED ISSUE #64984: qutebrowser: This application failed to start because no Qt platform plugin could be initialized.
## Issue description

`qutebrowser` aborts and produces the following error:

```
16:55:22 WARNING: Could not find the Qt platform plugin "xcb" in ""
16:55:22 CRITICAL: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Fatal Python error: Aborted

Current thread 0x00007ff93979eb80 (most recent call first):
  File "/nix/store/i34zmpg99l9b21qjg78alqj8vzh7q6zl-qutebrowser-1.6.3/lib/python3.7/site-packages/qutebrowser/misc/earlyinit.py", line 87 in _die
  File "/nix/store/i34zmpg99l9b21qjg78alqj8vzh7q6zl-qutebrowser-1.6.3/lib/python3.7/site-packages/qutebrowser/misc/earlyinit.py", line 222 in _check_modules
  File "/nix/store/i34zmpg99l9b21qjg78alqj8vzh7q6zl-qutebrowser-1.6.3/lib/python3.7/site-packages/qutebrowser/misc/earlyinit.py", line 238 in check_libraries
  File "/nix/store/i34zmpg99l9b21qjg78alqj8vzh7q6zl-qutebrowser-1.6.3/lib/python3.7/site-packages/qutebrowser/misc/earlyinit.py", line 296 in early_init
  File "/nix/store/i34zmpg99l9b21qjg78alqj8vzh7q6zl-qutebrowser-1.6.3/lib/python3.7/site-packages/qutebrowser/qutebrowser.py", line 190 in main
  File "/nix/store/i34zmpg99l9b21qjg78alqj8vzh7q6zl-qutebrowser-1.6.3/bin/..qutebrowser-wrapped-wrapped", line 11 in <module>
```
I suspect it's due to #54525, but when I started building a NixOS vm with the merge commit checked out, Nix started building `cmake`; I just don't have the computing resources to build from such a starting point.

What I do know is that the problem exists in ef6f912245b but not in 696767a9c9dc

### Steps to reproduce

1. Run `qutebrowser`

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.58, NixOS, 19.09.git.ef6f912245b (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.09pre172038.5d3fd3674a6"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### LINKED ISSUE #65290: qutebrowser doesn't start
**Describe the bug**
qutebrowser doesn't start

**To Reproduce**
1. Use the latest nixos-unstable
2. Start qutebrowser

**Expected behavior**
qutebrowser starts

**Screenshots**

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.60, NixOS, 19.09pre186326.62cb4f72282 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(philipp): `""`
 - channels(root): `""`
 - nixpkgs: `/nix/store/82bi282j8b6xx1gksqxjvrmwqh1mzwh4-source`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by emmanuelrosa**
This commit fixes the error:

Could not find the Qt platform plugin "xcb" in ""

and upgrades qutebrowser to 1.7.0

Closes #64984
Closes #65290
Closes #65150 

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Are we sure `wrapGAppsHook` is needed here?

I'm assuming it was there before because it wraps `XDG_DATA_DIRS`, and `wrapQtAppsHook` does that so it should be unneeded.

**Comment by emmanuelrosa** [comment]
With wrapGAppsHook, the wrapper looks like this:

```
#! /nix/store/xfghy8ixrhz3kyy6p724iv3cxji088dx-bash-4.4-p23/bin/bash -e
export GIO_EXTRA_MODULES='/nix/store/r78d406bbjbb6ziwqgyvxs8si25cadrq-glib-networking-2.60.3/lib/gio/modules'${GIO_EXTRA_MODULES:+':'}$GIO_EXTRA_MODULES
export XDG_DATA_DIRS='/nix/store/inlknqkinymvskvigpj345mfyfdvichx-gtk+3-3.24.10/share/gsettings-schemas/gtk+3-3.24.10:/nix/store/r3g9yn2cw6239szfq65q2cxw4cg35a81-gsettings-desktop-schemas-3.32.0/share/gsettings-schemas/gsettings-desktop-schemas-3.32.0'${XDG_DATA_DIRS:+':'}$XDG_DATA_DIRS
export XDG_DATA_DIRS='/nix/store/81k455cvnlfcf3an8zs4y41vwi0g2vyd-qutebrowser-1.6.3/share'${XDG_DATA_DIRS:+':'}$XDG_DATA_DIRS
export GST_PLUGIN_SYSTEM_PATH_1_0='/nix/store/nr5hv7dklhgxd684lsqqbxf7sh7w4zf5-gstreamer-1.16.0/lib/gstreamer-1.0:/nix/store/ynkfgyhz7jghvy524y0xxypx2fn9pfy6-gst-plugins-base-1.16.0/lib/gstreamer-1.0:/nix/store/0d1x359zfyjc2j9snf9dym3avkxbqbnj-gst-plugins-good-1.16.0/lib/gstreamer-1.0:/nix/store/gl82wwf8jkjbln6ivdam760phdiang7f-gst-plugins-bad-1.16.0/lib/gstreamer-1.0:/nix/store/xg67dk9vwzsvdgd5ks20ikckic4j82lr-gst-plugins-ugly-1.16.0/lib/gstreamer-1.0:/nix/store/xmv4gcr8jljch61zca551f84dkmxkrci-gst-libav-1.16.0/lib/gstreamer-1.0'${GST_PLUGIN_SYSTEM_PATH_1_0:+':'}$GST_PLUGIN_SYSTEM_PATH_1_0
export QT_PLUGIN_PATH='/nix/store/qa3faa7dy41wxd1y2ldmg5fzc3pdwb10-qtbase-5.12.0-bin/lib/qt-5.12/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QT_PLUGIN_PATH='/nix/store/js4fqglx52l2gb8f1ad34r8b11xj3y5w-qtsvg-5.12.0-bin/lib/qt-5.12/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QT_PLUGIN_PATH='/nix/store/xvifggs4vc2czi95gir0gpvm45ysin3s-qtdeclarative-5.12.0-bin/lib/qt-5.12/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QML2_IMPORT_PATH='/nix/store/xvifggs4vc2czi95gir0gpvm45ysin3s-qtdeclarative-5.12.0-bin/lib/qt-5.12/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QML2_IMPORT_PATH='/nix/store/319j582m94yaar88d6qk03rgm658l26f-qtquickcontrols-5.12.0/lib/qt-5.12/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QT_PLUGIN_PATH='/nix/store/wn784aaax0q85l5inf71ayngs14sli61-qtwayland-5.12.0-bin/lib/qt-5.12/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QML2_IMPORT_PATH='/nix/store/wn784aaax0q85l5inf71ayngs14sli61-qtwayland-5.12.0-bin/lib/qt-5.12/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export XDG_DATA_DIRS='/nix/store/81k455cvnlfcf3an8zs4y41vwi0g2vyd-qutebrowser-1.6.3/share'${XDG_DATA_DIRS:+':'}$XDG_DATA_DIRS
exec -a "$0" "/nix/store/81k455cvnlfcf3an8zs4y41vwi0g2vyd-qutebrowser-1.6.3/bin/.qutebrowser-wrapped_"  "${extraFlagsArray[@]}" "$@"
```

Without wrapGAppsHook, the wrapper looks like this:

```
#! /nix/store/xfghy8ixrhz3kyy6p724iv3cxji088dx-bash-4.4-p23/bin/bash -e
export QT_PLUGIN_PATH='/nix/store/qa3faa7dy41wxd1y2ldmg5fzc3pdwb10-qtbase-5.12.0-bin/lib/qt-5.12/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QT_PLUGIN_PATH='/nix/store/js4fqglx52l2gb8f1ad34r8b11xj3y5w-qtsvg-5.12.0-bin/lib/qt-5.12/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QT_PLUGIN_PATH='/nix/store/xvifggs4vc2czi95gir0gpvm45ysin3s-qtdeclarative-5.12.0-bin/lib/qt-5.12/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QML2_IMPORT_PATH='/nix/store/xvifggs4vc2czi95gir0gpvm45ysin3s-qtdeclarative-5.12.0-bin/lib/qt-5.12/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QML2_IMPORT_PATH='/nix/store/319j582m94yaar88d6qk03rgm658l26f-qtquickcontrols-5.12.0/lib/qt-5.12/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QT_PLUGIN_PATH='/nix/store/wn784aaax0q85l5inf71ayngs14sli61-qtwayland-5.12.0-bin/lib/qt-5.12/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QML2_IMPORT_PATH='/nix/store/wn784aaax0q85l5inf71ayngs14sli61-qtwayland-5.12.0-bin/lib/qt-5.12/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export XDG_DATA_DIRS='/nix/store/i7kqq9zfrw6bvm02gq5xp1vpkj76b858-qutebrowser-1.6.3/share'${XDG_DATA_DIRS:+':'}$XDG_DATA_DIRS
exec -a "$0" "/nix/store/i7kqq9zfrw6bvm02gq5xp1vpkj76b858-qutebrowser-1.6.3/bin/.qutebrowser-wrapped_"  "${extraFlagsArray[@]}" "$@"

```

A brief smoke-test didn't reveal any issues at runtime, but I want to test it further before committing to removing wrapGAppsHook.

**Comment by worldofpeace** [comment]
Ah, it would be needed for `glib-networking` and `gstreamer` which I didn't check.

**Comment by worldofpeace** [comment]
Think this should be good, do you want to combine this with a change like #65150 @emmanuelrosa  ?

File: pkgs/applications/networking/browsers/qutebrowser/default.nix [hunk 1]
```diff
@@ -95,6 +98,12 @@ in python3Packages.buildPythonApplication rec {
     done
   '';
 
+  postFixup = ''
+    wrapProgram $out/bin/qutebrowser \
+      "''${gappsWrapperArgs[@]}" \
+      "''${qtWrapperArgs[@]}" \
```
Comment (worldofpeace) [inline pkgs/applications/networking/browsers/qutebrowser/default.nix]:
```suggestion
      "''${qtWrapperArgs[@]}"
```

**Comment by emmanuelrosa** [comment]
Yes, @worldofpeace 

I'll go ahead and apply the qutebrowser-1.7.0 commit

**Review by worldofpeace — APPROVED** [review]
Runs for me in NixOS in a non-qt environment.

**Pull request merged**